programs/bpf_loader/src/lib.rs

```diff
@@ -162,12 +163,45 @@ pub fn process_instruction(
     instruction_data: &[u8],
     invoke_context: &mut dyn InvokeContext,
 ) -> Result<(), InstructionError> {
-    solana_logger::setup_with_default("solana=info");
-
+    process_instruction_with_explicit_lifetimes(
```

Comment:
Doing these gymnastic tricks to get make the lifetime's work when sharing InvokeContext in an Rc<RefCell>> between `info!` and `create_vm`. 

Please help!

Reply:
Yikes! Seems to me that a logger shouldn't be causing so much trouble. How about `Rc` wrapping its internals so that it can be freely cloned and have a separate lifetime from `InvokeContext`? Could maybe add a `logger() -> LogCollector` method to the InvokeContext trait?

Reply:
@jstarry Or maybe InvokeContext comes bundled in an `Rc<RefCell>>`?

Reply:
Ha, I was going to suggest that too but it seemed like a bigger lift, sgtm!